### DISCUSSION TITLE
Web hook handler needs to return a success response

### DISCUSSION THREAD

**Discussion by AminRafaey** [kinngh/shopify-node-mongodb-next-app#58] [General]
Hi @kinngh, Shopify has a rule that if your server doesn't return a response or return an error. Shopify will keep hitting your provided URL for webhook. So in the mentioned function shouldn't we return `res.status(200).send()`?
https://github.com/kinngh/shopify-nextjs-prisma-app/blob/21dcce9cfe673f17c4187c7cc39438c96f31d8e2/utils/webhooks/app_uninstalled.js#L10

**Comment by AminRafaey** [comment]
If you want I can create a PR for that. If I am not wrong this function is not returning anything and Shopify will hit our webhook 19 times to make sure we have received a request on our server successfully.

**Reply by kinngh** [reply]
While I appreciate the enthusiasm,
1. I don't take contributions / PRs to the repos
2. This is handled by `/pages/api/webhooks/[...webhookTopic].js`, which is the actual API endpoint. All the other places are more about what functions need to run when that endpoint is called.
3. `shopify.webhooks.process()` handles the return status automatically. Ref: https://github.com/Shopify/shopify-app-js/blob/main/packages/apps/shopify-api/docs/reference/webhooks/process.md
4. Please read the docs for the API libraries too, instead of just shopify.dev
